src/components/tools/NewToolRequestDialog.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * @author sriram
+ */
+import React, { useState } from "react";
+
+import { useTranslation } from "i18n";
+import { useMutation } from "react-query";
+
+import { Field, Form, Formik } from "formik";
+
+import DEDialog from "components/utils/DEDialog";
+import ErrorTypographyWithDialog from "components/utils/error/ErrorTypographyWithDialog";
+import ids from "./ids";
+import constants from "./constants";
+import { toolRequest } from "serviceFacades/tools";
+
+import {
+    build,
+    announce,
+    FormMultilineTextField,
+    FormTextField,
+} from "@cyverse-de/ui-lib";
+
+import Button from "@material-ui/core/Button";
+
+export default function NewToolRequestDialog(props) {
+    const { open, baseId, onClose } = props;
+    const [requestError, setRequestError] = useState();
+
+    const { t } = useTranslation("tools");
+
+    const [submitNewToolRequest, { status: submitRequestStatus }] = useMutation(
+        ({ submission }) => toolRequest(submission),
+        {
+            onSuccess: (data) => {
+                announce({
+                    text: t("toolRequestSuccess"),
+                });
+                onClose();
+            },
+            onError: setRequestError,
+        }
+    );
+
+    const handleSubmit = (values, { props }) => {
+        //test_data_path and cmd_line are required by service. Most of the times those info. can be
+        // found in the source repo. So copy the source url to those fields if not provided by user.
+        if (!values.test_data_path) {
+            values.test_data_path = values.source_url;
+        }
+        if (!values.cmd_line) {
+            values.cmd_line = values.source_url;
+        }
+        const submission = { ...values };
```

Comment:
Good comments 👍 
I would not modify the form's values, though, but modify the `submission` copy instead:
```suggestion
        const submission = { ...values };
        
        //test_data_path and cmd_line are required by service. Most of the times those info. can be
        // found in the source repo. So copy the source url to those fields if not provided by user.
        if (!submission.test_data_path) {
            submission.test_data_path = submission.source_url;
        }
        if (!submission.cmd_line) {
            submission.cmd_line = submission.source_url;
        }
```